packages/backend/src/utils/progressiveEvent.ts

```diff
@@ -0,0 +1,50 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export interface ProgressiveEvent {
```

Comment:
I don't understand why you renamed from `ProgressEvent` to `ProgressiveEvent` ?

Reply:
I feel like `Progress` and `Progressive` are too close and could lead to some confusion, we might want to find something different to improve readability.

`BaseEvent` instead of `DownloadEvent` and keep `ProgressEvent` to talk about the real progress.

or

`TransferEvent` if we want to keep some _action/wording_ of what is happening 

> I have a preference for BaseEvent, as we could create its own `IBaseEvent` class, and keep it separate from the logic of the Downloader/Uploader

Reply:
progressing = happening or developing gradually or in stages. Look to me that it expresses well what it does and it can be reused. 
By using TrasferEvent we are binding it only to transferring actions which may not be true in future, no?
Maybe `ExecutingProgressiveEvent` instead of `ProgressProgressiveEvent`?

Reply:
here is the `Download` event

```
export interface DownloadEvent {
  id: string;
  status: 'error' | 'completed' | 'progress' | 'cancelled';
  message?: string;
}
```

This seems very _basic_ and generic, I still think something _short_ and easy to read like `BaseEvent` would be more suitable than `ExecutingProgressiveEvent`...